src/target/riscv/riscv.c

```diff
@@ -1614,6 +1614,441 @@ static int riscv_hit_trigger_hit_bit(struct target *target, uint32_t *unique_id)
+			imm = get_imm_cldsp(instruction);
+		} else { // MATCH_C_FLWSP
+			imm = get_imm_clwsp(instruction);
+		}
+		break;
+
+	case MATCH_C_FLDSP: // or MATCH_C_LQSP if RV128C
+		if (riscv_xlen(target) == 128) { // MATCH_C_LQSP
+			imm = get_imm_clqsp(instruction);
+		} else { // MATCH_C_FLDSP
+			imm = get_imm_cldsp(instruction);
+		}
+		break;
+
+	case MATCH_C_SWSP:
+		imm = get_imm_cswsp(instruction);
+		break;
+
+	case MATCH_C_SDSP: // if xlen >= 64 or MATCH_C_FSWSP:
+		if (riscv_xlen(target) > 32) { // MATCH_C_SDSP
+			imm = get_imm_csdsp(instruction);
+		} else { // MATCH_C_FSWSP
+			imm = get_imm_cswsp(instruction);
+		}
+		break;
+
+	case MATCH_C_FSDSP: // or MATCH_C_SQSP if xlen == 128
+		if (riscv_xlen(target) == 128) { // MATCH_C_SQSP
+			imm = get_imm_csqsp(instruction);
+		} else { // MATCH_C_FSDSP
+			imm = get_imm_csdsp(instruction); // same as C.SDSP
+		}
+		break;
+
+	case MATCH_C_LW:
+		imm = get_imm_clw(instruction);
+		break;
+
+	case MATCH_C_FLW: // or MATCH_C_LD if xlen >= 64
+		if (riscv_xlen(target) > 32) { // MATCH_C_LD
+			imm = get_imm_cld(instruction);
+		} else { // MATCH_C_FLW
+			imm = get_imm_clw(instruction); // same as C.FLW
+		}
+		break;
+
+	case MATCH_C_FLD: // or MATCH_C_LQ if xlen == 128
+		if (riscv_xlen(target) == 128) { // MATCH_C_LQ
+			imm = get_imm_clq(instruction);
+		} else { // MATCH_C_FLD
+			imm = get_imm_cld(instruction); // same as C.LD
+		}
+		break;
+
+	case MATCH_C_SW:
+		imm = get_imm_clw(instruction); // same as C.LW
+		break;
+
+	case MATCH_C_FSW: // or MATCH_C_SD if xlen >= 64
+		if (riscv_xlen(target) > 32) { // MATCH_C_SD
+			imm = get_imm_cld(instruction); // same as C.LD
+		} else { // MATCH_C_FSW
+			imm = get_imm_clw(instruction); // same as C.LW
+		}
+		break;
+
+	case MATCH_C_FSD: // or MATCH_C_SQ if xlen == 128
+		if (riscv_xlen(target) == 128) { // MATCH_C_SQ
+			imm = get_imm_clq(instruction); // same as C.LQ
+		} else { // MATCH_C_FSD
+			imm = get_imm_cld(instruction); // same as C.LD
+		}
+		break;
+
+	default:
+		LOG_TARGET_DEBUG(target, "%x is not a RV32I or \"C\" load or store",
+						 instruction);
+		return ERROR_FAIL;
+	}
+	*memoffset = imm;
+	return ERROR_OK;
+}
+
+static inline int verify_storeload(struct target * target,
+		const riscv_insn_t instruction, const target_addr_t mem_addr) {
+	uint32_t opcode = get_opcode(instruction);
+	bool misa_f = riscv_supports_extension(target, 'F');
+	bool misa_d = riscv_supports_extension(target, 'D');
+
+	switch (opcode) {
+	case MATCH_LB:
+	case MATCH_FLH & ~INSN_FIELD_FUNCT3:
+	case MATCH_SB:
+	case MATCH_FSH & ~INSN_FIELD_FUNCT3:
+		if (opcode == MATCH_SB || opcode == (MATCH_FSH & ~INSN_FIELD_FUNCT3)) {
+			LOG_TARGET_DEBUG(target, "%x is store instruction", instruction);
+		} else if (opcode == MATCH_LB ||
+				   opcode == (MATCH_FLH & ~INSN_FIELD_FUNCT3)) {
+			LOG_TARGET_DEBUG(target, "%x is load instruction", instruction);
+		} else {
+			LOG_TARGET_DEBUG(target, "Unknown instruction");
+			return ERROR_FAIL;
```

Comment:
[same as above] AFAIU, this block is unreachable. Consider replacing the return with an assertion.
```suggestion
			assert(false);
```